### ISSUE TITLE
Browser input upload after Drag and Drop issue

### ISSUE THREAD

**Issue by flo78** [manolo/gwtupload#5]
I use a SingleUploader instantiated like this with drag and drop and custom upload link
```
   // Single Uploader / Upload button 
	    Label uploadButton = new Label();
	    uploadButton.setText("upload");
	    // uploader
	    FileInputType input = FileInputType.CUSTOM.with(uploadButton).with((IDropZone) customExternalDropZone);
	    SingleUploader dragDropUploader = new SingleUploader(input,new UploadStatus());
 ```
Case that works:
Single upload of one or multiple file with classical browser input form
Single Drag and drop  of one or multiple file.
Single Drag and drop  of one or multiple file after making a single upload of one or multiple file with classical browser input form

What does not work :
Upload of one or multiple file with classical browser input form after making a single Drag and drop  of one or multiple file

Investigating the code,I found in the class DragAndDropFormPanel , that the browser input in not detected and drag and drop is detected (probably due to past drag and drop):
```
    @Override
    public void submit() {
        final List<IFileInput> childFileInputs = getChildFileInputs();
        if (childFileInputs == null || childFileInputs.isEmpty()) {
            return;
        }
        final ArrayList<IDragAndDropFileInput> dndFileInputs = 
                new ArrayList<IDragAndDropFileInput>();
        boolean thereAreNonDragAndDropFileInputs = false;
        for (IFileInput fileInput : childFileInputs) {
            if (fileInput instanceof IDragAndDropFileInput) {
                final IDragAndDropFileInput dndFileInput = (IDragAndDropFileInput) fileInput;
                if (dndFileInput.thereAreDragAndDropedFiles()) {
                    dndFileInputs.add(dndFileInput);
                } else {
                    // files might be selected via file dialog
                    thereAreNonDragAndDropFileInputs = true;
                }
            } else {
                // files are selected via file dialog
                thereAreNonDragAndDropFileInputs = true;
            }
        }
        if (thereAreNonDragAndDropFileInputs) {
            super.submit(); // submit non-dnd file inputs
        }
        if (!dndFileInputs.isEmpty()) {
            submitDragAndDropFileInputs(dndFileInputs); // submit dnd file inputs
        }
    }
```
because there were drag and dropped file in the past, the call dndFileInput.thereAreDragAndDropedFiles() is always returning true, even if I make a classical browser input upload instead of a drag and drop upload. I cannot find a place where the list of drag and dropped files is cleared.

**Comment by manolo** [comment]
Fixed, use 1.0.2-SNAPSHOT until 1.0.2 is released.
You can check if the issue is fixed in the project example, use the 4 example at http://gwtupload.alcala.org/gupld/gwtuploadsample.SingleUploadSample/SingleUploadSample.html 

Thanks for reporting

**Comment by flo78** [comment]
thx for the fix :+1:
